frontend/app/(tabs)/profile/profileEditModal.tsx

```diff
@@ -0,0 +1,129 @@
+import UserInput from '@/components/UserInput';
+import { UserState } from '@/types/user';
+import { isClerkAPIResponseError, useAuth, useUser } from '@clerk/clerk-expo';
+import { MediaTypeOptions, launchImageLibraryAsync } from 'expo-image-picker';
+import { Stack } from 'expo-router';
+import { Controller, useForm } from 'react-hook-form';
+import { TouchableOpacity } from 'react-native';
+import { Avatar, Separator, View, YStack, Form, XStack, Button } from 'tamagui';
+
+const ProfileEditModal = () => {
+  const { user } = useUser();
+
+  const {
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<UserState>({
+    defaultValues: {
+      username: user?.username,
+      firstName: user?.firstName,
+      lastName: user?.lastName,
+      password: user?.passwordEnabled ? 'password' : '',
+    },
+  });
+
+  const handleUserProfileChange = async () => {
```

Comment:
This feature is off to a great start! I like being able to add a profile picture and editing my bio.